### PR TITLE
fix(Tearsheet): implement focus trapping within tearsheets in V1

### PR DESCRIPTION
**Pull request by makafsal**
Contributes to #3499 

change in focus claim function for stacked tearsheets
handle tab key press operation properly to navigate through fields

V2 PR: https://github.com/carbon-design-system/ibm-products/pull/4129

### CODE REVIEW

**Comment by makafsal** [comment]
check

**Comment by makafsal** [comment]
recheck

**Review by amal-k-joy — DISMISSED** [review]
Code looks fine and seems to be working.

**Review by davidmenendez — APPROVED** [review]

**Comment by makafsal** [comment]
Implemented the `useFocus()` hook in this branch as well.
Path: `packages/ibm-products/src/global/js/hooks/useFocus.js`

File: packages/ibm-products/src/global/js/hooks/useFocus.js [hunk 1]
```diff
@@ -0,0 +1,80 @@
+/**
+ * Copyright IBM Corp. 2020, 2023
```
Comment (elycheea) [inline packages/ibm-products/src/global/js/hooks/useFocus.js:2]:
```suggestion
 * Copyright IBM Corp. 2024, 2024
```

**Review by davidmenendez — APPROVED** [review]

**Review by davidmenendez — APPROVED** [review]

**Pull request merged**